app/uk/gov/hmrc/economiccrimelevyreturns/controllers/RelevantApLengthController.scala

```diff
@@ -53,19 +58,66 @@ class RelevantApLengthController @Inject() (
   }
 
   def onSubmit(mode: Mode): Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
+    implicit val hc: HeaderCarrier = CorrelationIdHelper.getOrCreateCorrelationId(request)
     form
       .bindFromRequest()
       .fold(
         formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, request.startAmendUrl))),
-        relevantApLength =>
-          eclReturnsConnector
-            .upsertReturn(
-              dataCleanup.cleanup(request.eclReturn.copy(relevantApLength = Some(relevantApLength)))
+        relevantApLength => {
+          val eclReturn = dataCleanup.cleanup(request.eclReturn.copy(relevantApLength = Some(relevantApLength)))
+          eclReturnsService
+            .upsertReturn(eclReturn)
+            .asResponseError
+            .foldF(
+              error => Future.successful(Status(error.code.statusCode)(Json.toJson(error))),
```

Comment:
How does this error message manifest? We should be showing the default error page, but this looks like it'll show JSON like what's happening in Account server.